### PR TITLE
Refactor: Moved Search to Header

### PR DESCRIPTION
**Pull request by priscilawebdev**
Type: Refactor

Description: "Add search bar in the sidebar. It will not be displayed in mobile version "
Task: #1018

* Done:
 - [x] Moved Search to Header component
 - [x] Added AutoComplete funcionality
 - [x] Added Mobile Version

* To do:
 - [ ] **Tests (Attention please)**
 - [ ] Remove inline css (2x). I will create something reusable.
-  [ ] Refactor App (got large... I need to split it up), Home and split some methods.

### CODE REVIEW

**Comment by juanpicado** [comment]
After merge https://github.com/verdaccio/verdaccio/pull/1067 you can start getting rid of `bind(this)` syntax.

File: src/webui/app.js [hunk 1]
```diff
@@ -118,48 +187,159 @@ export default class App extends Component {
...
+    });
+  };
+
+  handleKeyDown = event => {
+    if (event.key === 'Enter') {
+      const { filteredPackages, packages } = this.state;
+      const value = event.target.value.trim();
+      this.setState({
+        filteredPackages: value ? 
+        packages.filter(pkage => pkage.label.match(value)) : filteredPackages
```
Comment (ayusharma) [inline src/webui/app.js]: better to use `pkg` instead of `pkage`

File: src/webui/app.js [hunk 2]
```diff
@@ -22,30 +32,46 @@ export default class App extends Component {
...
-    this.handleLogout = this.handleLogout.bind(this);
-    this.toggleLoginModal = this.toggleLoginModal.bind(this);
-    this.doLogin = this.doLogin.bind(this);
-    this.loadLogo = this.loadLogo.bind(this);
-    this.isUserAlreadyLoggedIn = this.isUserAlreadyLoggedIn.bind(this);
+    isUserLoggedIn: false,
+    packages: [],
+    searchPackages: [],
+    filteredPackages: [],
+    search: "",
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply src/webui/app.js]: done!

File: src/webui/app.js (hunk 1 above)
Comment (priscilawebdev) [reply src/webui/app.js]: done!

**Comment by priscilawebdev** [comment]
@ayusharma @juanpicado tests fixed 😄

**Pull request merged**